main.py

```diff
@@ -1,255 +1,111 @@
-
-        delivery.ack()
-        self.logger.debug("published message %s",  delivery)
-        wes_data_service_messages_published_total.inc()
-
-    def new_pod_state(self):
-        return PodState(pod=None, backlog=[], updated_at=self.clock.now())
-
-    def clear_backlogs_but_keep_pod_state(self):
-        for pod_state in self.pod_state.values():
-            pod_state.backlog.clear()
-        wes_data_service_messages_in_backlog.set(0)
-
-
-class Publisher:
 
-    logger = logging.getLogger("Publisher")
-
-    def __init__(self, params):
-        self.params = params
-        self._connect()
-
-    def _connect(self):
-        self.logger.debug("publisher connecting...")
-        self.connection = pika.BlockingConnection(self.params)
-        self.channel = self.connection.channel()
-        self.logger.debug("publisher connected")
+        # handle upload message case: needs to have value changed to url
+        if msg.name == "upload":
+            msg = convert_to_upload_message(msg, self.upload_publish_name)
 
-    def close(self):
-        self.channel.close()
-        self.connection.close()
+        # update message app meta
+        app_meta = self.app_meta_cache.get(app_uid)
 
-    def publish(self, exchange, routing_key, publishing):
-        while True:
-            try:
-                return self.channel.basic_publish(
-                    exchange=exchange,
-                    routing_key=routing_key,
-                    properties=publishing.properties,
-                    body=publishing.body,
-                )
-            except Exception:
-                if self.logger.isEnabledFor(logging.DEBUG):
-                    self.logger.exception("publisher failed to publish. will reconnect and retry...")
-                time.sleep(3.0)
-                self._connect()
+        if app_meta is None:
+            self.logger.info("reject msg: no pod meta: %r", msg)
+            ch.basic_reject(method.delivery_tag, False)
+            wes_data_service_messages_rejected_total.inc()
+            return
 
+        for k, v in app_meta.items():
+            msg.meta[k] = v
 
-def update_message_with_config_metadata(msg: wagglemsg.Message, config: MessageHandlerConfig):
-    msg.meta["node"] = config.node
-    msg.meta["vsn"] = config.vsn
+        # update message system meta
+        for k, v in self.system_meta.items():
+            msg.meta[k] = v
 
+        self.publish_message(ch, method.routing_key, msg)
+        ch.basic_ack(method.delivery_tag)
+    
+    def publish_message(self, ch, routing_key: str, msg: wagglemsg.Message):
+        body = wagglemsg.dump(msg)
 
-def update_message_with_pod_metadata(msg: wagglemsg.Message, pod):
-    # add scheduler metadata using pod uid
-    msg.meta["host"] = pod.host
-    # TODO include namespace
-    msg.meta["plugin"] = pod.image.split("/")[-1]
-    msg.meta["job"] = pod.labels.get("sagecontinuum.org/plugin-job", "sage")
+        if routing_key in ["node", "all"]:
```

Comment:
any chance these hard-coded values "node" and "all" (and "beehive" below) can be variables in some way?

Reply:
Yes, I can do that. Just note that these are kind of implicitly dependent on apps / pywaggle too. But I'll move them to constants so it's easier to find.

Reply:
I consolidated these to SCOPE_ "constants" in main.py